src/plugins/util/OverlayPlugin.js

```diff
@@ -91,4 +93,12 @@ module.exports = class OverlayPlugin {
             this.restoreState();
         }
     }
+
+    toggleWidget(widget,toolbarBtnImage) {
+        this.instance.store.dispatch({type: 'OVERLAY_OPEN', payload: widget.classList.contains('gm-hidden')});
+
+        // Toggle display
+        widget.classList.toggle('gm-hidden');
```

Comment:
maybe instead of passing the widget as an arg we could iterate over `this.overlays`, like it's done in the `closeOverlays` function? (also maybe this new function and `closeOverlays` should be refactored and merged)